stan/math/gpu/set_kernel_args.hpp

```diff
@@ -0,0 +1,25 @@
+#ifndef STAN_MATH_GPU_SET_KERNEL_ARGS_HPP
+#define STAN_MATH_GPU_SET_KERNEL_ARGS_HPP
+#ifdef STAN_OPENCL
+#include <CL/cl.hpp>
+
+namespace stan {
+namespace math {
+inline void _set_kernel_args(cl::Kernel &k, int i) {}  // terminating function
```

Comment:
what does this do?

Please change the name... names can't start with a leading underscore (according to spec). https://stackoverflow.com/questions/228783/what-are-the-rules-about-using-an-underscore-in-a-c-identifier

Reply:
@SteveBronder Steve can you take a look at this. I am not really sure here.

Reply:
> All identifiers that begin with an underscore are always reserved for use as identifiers with file scope in both the ordinary and tag name spaces.

`_set_kernel_args()` is only used within file scope so this is fine, no? Totally open to changing the name. Will have to think about what to call it tho'

Reply:
Did not see this comment earlier so I am pasting from below
__________________________________
`set_kernel_args()` is a variadic template'd function that helps remove some boiler plate code.

When we pass the kernel arguments instead of writing
```cpp
kernel.setArg(0, boo)
kernel.setArg(1, moo)
kernel.setArg(2, foo)
kernel.setArg(3, doo)
```
we can write

```cpp
set_kernel_args(kernel, boo, moo, foo, doo)
```